core/src/services/fs/writer.rs

```diff
@@ -101,3 +105,23 @@ impl oio::BlockingWrite for FsWriter<std::fs::File> {
         Ok(())
     }
 }
+
+impl oio::PositionWrite for FsWriter<tokio::fs::File> {
+    async fn write_all_at(&self, offset: u64, buf: Buffer) -> Result<()> {
```

Comment:
Hi, write_all_at should consume the entire `buf`. But `buf.chunk()` only returns the first slice.

Reply:
> @Xuanwo Windows doesn't have a `write_all_at` method.

Hi, we can call `write_at` in loop to implement `write_all_at`.

Reply:
But If we don't have a `abort` method? we can't pass through `behavior::test_writer_abort`.

Reply:
> But If we don't have a `abort` method? we can't pass through `behavior::test_writer_abort`.

We need `write_all_at`, `close`, and `abort`. I previously disagreed with using `write_at` because it complicates implementing concurrent writing.

Reply:
Plz review.